From 6ba0619aeccd979925252ec6f3dfde8c04fe1390 Mon Sep 17 00:00:00 2001
From: "Wong, Vincent Por Yin" <vincent.por.yin.wong@intel.com>
Date: Mon, 20 Jul 2020 16:10:46 +0800
Subject: [PATCH 2/3] REVERTME: net: stmmac: use DMA Bus Mode INTM MODE 0

Originally, this was done for TGL platforms only, this patch
sets DMA_BUS_MODE_INTM_MODE0 for all dwmac4 devices. This works
around the multi queue receiving issue for EHL when TCC is
disabled.

Signed-off-by: Wong, Vincent Por Yin <vincent.por.yin.wong@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/dwmac4_dma.c | 10 +++-------
 drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c |  3 ---
 include/linux/stmmac.h                           |  1 -
 3 files changed, 3 insertions(+), 11 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/dwmac4_dma.c b/drivers/net/ethernet/stmicro/stmmac/dwmac4_dma.c
index 9bef3d8d7423..cfdd9367ab02 100644
--- a/drivers/net/ethernet/stmicro/stmmac/dwmac4_dma.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac4_dma.c
@@ -165,13 +165,9 @@ static void dwmac4_dma_init(void __iomem *ioaddr,
 
 	value = readl(ioaddr + DMA_BUS_MODE);
 
-	if (dma_cfg->multi_msi_en && dma_cfg->tgl_wa) {
-		value &= ~DMA_BUS_MODE_INTM_MASK;
-		value |= (DMA_BUS_MODE_INTM_MODE0 << DMA_BUS_MODE_INTM_SHIFT);
-	} else if (dma_cfg->multi_msi_en) {
-		value &= ~DMA_BUS_MODE_INTM_MASK;
-		value |= (DMA_BUS_MODE_INTM_MODE1 << DMA_BUS_MODE_INTM_SHIFT);
-	}
+	value &= ~DMA_BUS_MODE_INTM_MASK;
+	value |= (DMA_BUS_MODE_INTM_MODE0 << DMA_BUS_MODE_INTM_SHIFT);
+
 	writel(value, ioaddr + DMA_BUS_MODE);
 }
 
diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
index df95ac66c12e..559829a25220 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
@@ -538,9 +538,6 @@ static int tgl_common_data(struct pci_dev *pdev,
 	/* Maximum TX XDP queue */
 	plat->max_combined = 2;
 
-	/* WORKAROUND: TGL has to use DMA INTM 0 to avoid intermittent reset */
-	plat->dma_cfg->tgl_wa = 1;
-
 	/* TX and RX Marvell 88E2110 PHY latency (ns) */
 	plat->phy_tx_latency_10 = 6652;
 	plat->phy_tx_latency_100 = 1152;
diff --git a/include/linux/stmmac.h b/include/linux/stmmac.h
index f55ec48c3b9a..4b6b622cb3f0 100644
--- a/include/linux/stmmac.h
+++ b/include/linux/stmmac.h
@@ -94,7 +94,6 @@ struct stmmac_dma_cfg {
 	int mixed_burst;
 	bool aal;
 	bool multi_msi_en;
-	bool tgl_wa;
 };
 
 #define AXI_BLEN	7
-- 
2.17.0

